src/redux/books/booksSlice.js

```diff
@@ -1,41 +1,38 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { getAllApiBooks, addBooktoAPI, deleteBookFromAPI } from '../../api/api';
 
-const initialState = [
-  {
-    item_id: '1',
-    title: 'The Great Gatsby',
-    author: 'John Smith',
-    category: 'Fiction',
-  },
-  {
-    item_id: '2',
-    title: 'Anna Karenina',
-    author: 'Leo Tolstoy',
-    category: 'Fiction',
-  },
-  {
-    item_id: '3',
-    title: 'The Selfish Gene',
-    author: 'Richard Dawkins',
-    category: 'Nonfiction',
-  },
-];
+export const getBooks = createAsyncThunk('books/getBooks', async () => {
+  const response = await getAllApiBooks();
+  return response;
+});
```

Comment:
- [ ] Kindly refactor your code Make sure that an action creator called fetchBooks exists in the booksSlice , fetch() always returns a list, while . get() returns the first result, or None if there are no results please fix this